### PR TITLE
Solve word count bug in mod_feed + some code review

### PR DESCRIPTION
**Pull request by n9iels**
## About this PR

This PR solves a bug with the word count parameter of mod_feed. If you load a feed with HTML tags (for example: http://community.joomla.org/blogs/community.feed?type=rss) and set the word count to 20 there are way less than 20 words visible. The two things that cause this problem:
- `JHtml::_('string.truncate')` counts the characters, not the words of the string
- HTML tags like `<a href="http://link.nl">` are counted too

This PR change the text of the parameter to `Character count` and I also adds `strip_tags()` for the character count.

Other changes in this PR are code style and some change to make the code more readably and logic.
## Test instructions
1. Make a mod_feed module and use http://community.joomla.org/blogs/community.feed?type=rss as Feed URL, set the RTL Feed paramter on "No" and all other parameters on "Yes" for now
2. Set the "Word count" parameter on 20, and confirm the feed displays less than 20 words on the front-end
3. Apply the patch, and confirm the parameter is now called "Character limit" and the amount of characters is better now (Maybe still not 20 characters, because the function don't cut words in half)
4. Confirm all parameters are working like they use to do
5. Install an RTL language (Persion for example), and test the "RTL Feed" parameter for this language. If "RTL Feed" is set to no, even on RTL languages the feed must be displayed LTR

### CODE REVIEW

File: modules/mod_feed/mod_feed.php [hunk 1]
```diff
@@ -12,20 +12,7 @@
...
-{
-	echo '<div>';
-	echo JText::_('MOD_FEED_ERR_NO_URL');
-	echo '</div>';
-
-	return;
-}
-
-$feed = ModFeedHelper::getFeed($params);
+$feed 			 = ModFeedHelper::getFeed($params);
```
Comment (wilsonge) [inline modules/mod_feed/mod_feed.php]: Please can you use spaces to align here! Thanks :)

Comment (n9iels) [inline modules/mod_feed/mod_feed.php]: Changed :+1:

**Comment by Ruchiranga** [comment]
I have tested this item :red_circle: unsuccessfully on 61f6253dd4577841736b9bcb0305baa1e1e9554c<br /><br />### Tested successfully

* The amount of characters shown has become better. See 
   [Before applying the patch](http://i.xomf.com/zbwrk.png "") and
   [After applying the patch](http://i.xomf.com/qflvq.png "") .
* All other parameters except "RTL Feed" work the way they should.

### Tested unsuccessfully

* The parameter is now called ["Character Count"](http://i.xomf.com/fjnbq.png "") not "Character limit"
* After installing Persian and using it, even if the "RTL Feed" is set to no, [the feed is displayed RTL](http://i.xomf.com/xjvww.png "").
(I was expecting the feed to be left aligned. Correct me if what I understood from the term "displayed LTR" is wrong)<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8230">issues.joomla.org/joomla-cms/8230</a>.</sub>

**Comment by n9iels** [comment]
@Ruchiranga Thanks for testing! Yes you're right, the RTL function wasn't working, I used a wrong variable. I also changed the label of the parameter to `Charcter limit

**Comment by Ruchiranga** [comment]
I have tested this item :white_check_mark: successfully on 607a4408950898aa6f38b00f0636618d639f774e<br /><br />* The amount of characters shown has become better. See 
   [Before applying the patch](http://i.xomf.com/zbwrk.png "") and
   [After applying the patch](http://i.xomf.com/qflvq.png "").
* All other parameters including ["RTL Feed"](http://i.xomf.com/hsvzz.png "") work the way they should.
* The parameter is now called "[Character limit](http://i.xomf.com/gnrnq.png "")"<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8230">issues.joomla.org/joomla-cms/8230</a>.</sub>

**Comment by designbengel** [comment]
I´m not sure if this breaks something on existing sites because even if the wordcount didn´t work right before 
i don´t know if we can just change it to character limit?

Isn´t it better to fix the wordcount instead?
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8230">issues.joomla.org/joomla-cms/8230</a>.</sub>

**Comment by brianteeman** [comment]
Changing from word count to character count will break peoples sites. Imagine I have it set to 10 

### before
my news feed of approximately ten words excepting html bug

### after
my news fe


So with this change form word to character count it cannot be accepted.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8230">issues.joomla.org/joomla-cms/8230</a>.</sub>

**Comment by brianteeman** [comment]
If you want to update the PR to retain the word count then this can be tested otherwise I am afraid it will have to be closed<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8230">issues.joomla.org/joomla-cms/8230</a>.</sub>

**Comment by brianteeman** [comment]
Closed for the reason above<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8230">issues.joomla.org/joomla-cms/8230</a>.</sub>

**Pull request closed without merging**